components/dashboard/src/admin/UserSearch.tsx

```diff
@@ -0,0 +1,105 @@
+    const [searchTerm, setSearchTerm] = useState('');
+    const [searching, setSearching] = useState(false);
+    const [currentUser, setCurrentUserState] = useState<User|undefined>(undefined);
+
+    useEffect(() => {
+        const userId = location.pathname.split('/')[3];
+        if (userId) {
+            let user = searchResult.rows.find(u => u.id === userId);
+            if (user) {
+                setCurrentUserState(user);
+            } else {
+                getGitpodService().server.adminGetUser(userId).then(
+                    user => setCurrentUserState(user)
+                ).catch(e => console.error(e));
+            }
+        } else {
+            setCurrentUserState(undefined);
+        }
+    }, [location]);
+
+    if (!user || !user?.rolesOrPermissions?.includes('admin')) {
+        return <Redirect to="/"/>
+    }
+
+    if (currentUser) {
+        return <UserDetail user={currentUser}/>;
+    }
+
+    const search = async () => {
+        setSearching(true);
+        try {
+            const result = await getGitpodService().server.adminGetUsers({
+                searchTerm,
+                limit: 50,
+                orderBy: 'creationDate',
+                offset: 0,
+                orderDir: "desc"
+            });
+            setSearchResult(result);
+        } finally {
+            setSearching(false);
+        }
+    }
+    return <PageWithSubMenu subMenu={adminMenu} title="Users" subtitle="Search an manage all users.">
+        <div className="pt-8 flex">
+            <div className="flex justify-between w-full">
+                <div className="flex">
+                    <div className="py-4">
+                        <svg className={searching ? 'animate-spin' : ''} width="16" height="16" fill="none" xmlns="http://www.w3.org/2000/svg">
+                            <path fillRule="evenodd" clipRule="evenodd" d="M6 2a4 4 0 100 8 4 4 0 000-8zM0 6a6 6 0 1110.89 3.477l4.817 4.816a1 1 0 01-1.414 1.414l-4.816-4.816A6 6 0 010 6z" fill="#A8A29E" />
+                        </svg>
+                    </div>
+                    <input className="border-0" type="text" placeholder="Search Users" onKeyDown={(ke) => ke.key === 'Enter' && search() } onChange={(v) => { setSearchTerm(v.target.value) }} />
+                </div>
+                <button className="" disabled={searching} onClick={search}>Start Search</button>
+            </div>
+        </div>
+        <div className="flex flex-col space-y-2">
+            <div className="px-6 py-3 flex justify-between space-x-2 text-sm text-gray-400 border-t border-b border-gray-200">
+                <div className="w-1/12"></div>
+                <div className="w-6/12">Name</div>
+                <div className="w-5/12">Created</div>
+            </div>
+        </div>
+        {searchResult.rows.filter(u => u.identities.length > 0).map(u => <UserEntry user={u} />)}
```

Comment:
**thought**: I still find this empty data set on load slightly confusing but I understand this helps us avoid some performance costs. Let's leave it as is for now. 🎯

Reply:
We should possibly show an empty list place holder (later).